scripts/check_pytest_raises_hygiene.py

```
@@ -0,0 +1,237 @@
+    control-flow context (the function body is not executed at the
+    pytest.raises call site, so no unreachable-code claim applies).
+    """
+    blocks: list[list[ast.stmt]] = []
+    if isinstance(node, ast.With | ast.AsyncWith):
+        blocks.append(node.body)
+    elif isinstance(node, ast.For | ast.AsyncFor | ast.While):
+        blocks.append(node.body)
+        blocks.append(node.orelse)
+    elif isinstance(node, ast.If):
+        blocks.append(node.body)
+        blocks.append(node.orelse)
```

Ruff (0.15.11)

[warning] Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)